### DISCUSSION TITLE
post/list-and-download-github-repo-cs

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#60] [Announcements]
# post/list-and-download-github-repo-cs

Mark Heath's Development Blog

https://markheath.net/post/list-and-download-github-repo-cs

**Comment by johndemeyer** [comment]
Hello,
 i am new to C#  and the use of nuggets.

When i use the code i get a error:  
CS0246	The type or namespace name 'Jarray' could not be found (are you missing a using directive or an assembly reference?)	

Can u show me the total code?

greetings John

**Reply by markheath** [reply]
You need to reference the Newtonsoft.Json NuGet package. And then your IDE should offer to automatically add a using statement that will allow it to find JArray.

**Reply by johndemeyer** [reply]
have installed it, still the same error  :-)

**Reply by markheath** [reply]
Did you add a using statement? It's probably `using Newtonsoft.Json.Linq;` 
You will need to add using statements whenever you use classes from NuGet packages

**Comment by martinogithub** [comment]
I I have many files I see only 1000
Can I filter using file name? example: *.json
Thank you
